Controllers/StatusController.cs

```diff
@@ -47,5 +47,33 @@ public async Task<ActionResult<IEnumerable<GetDataDTO<Department>>>> GetRole()
                 .ToListAsync();
             return Ok(new GetDataDTO<Department>(role));
         }
+
+        [Authorize(Roles = "Warehouse, Transportation")]
+        [HttpGet("order")]
+        public async Task<ActionResult<IEnumerable<GetDataDTO<OrderStatus>>>> GetOrderStatus()
+        {
+            if (!IsValidApiRequest())
+            {
+                return ApiBadRequest("Invalid Headers!");
+            }
+
+            var user = await GetCurrentUser();
+            switch (user.Department)
+            {
+                case DepartmentId.Warehouse:
+                    var status = await Context.OrderStatus.Where(o => (int)o.Id < 4 || (int)o.Id == 6)
+                        .Select(o => new OrderStatus(o))
+                        .ToListAsync();
+                    return Ok(new GetDataDTO<OrderStatus>(status));
+                case DepartmentId.Transportation:
+                    var status2 = await Context.OrderStatus.Where(o => (int)o.Id > 3)
+                        .Select(o => new OrderStatus(o))
+                        .ToListAsync();
+                    return Ok(new GetDataDTO<OrderStatus>(status2));
```

Comment:
Maybe we could extract `status` variable and move it above `switch` sentence?

Reply:
Extracted